packages/landing/src/map.ts

```diff
@@ -53,115 +29,54 @@ export class Basemaps {
-            const view = this.map.getView();
-            view.setZoom(location.zoom);
-            view.setCenter(this.locationFromLonLat(location));
+            this.map.setZoom(location.zoom);
+            const coordinate = locationTransform(location, this.config.tileMatrix, MapboxTms);
+            this.map.setCenter([coordinate.lon, coordinate.lat]);
         });
     }
 
-    private locationFromLonLat(location: MapLocation): Coordinate {
-        return proj.transform(
-            [location.lon, location.lat],
-            UrlProjection,
-            this.config.tileMatrix.projection.toEpsgString(),
-        );
-    }
-
     private getLocationFromHash(): MapLocation {
         return { ...DefaultCenter[this.config.tileMatrix.identifier], ...WindowUrl.fromHash(window.location.hash) };
     }
 
     /** Import location and rendering information from the url */
     private updateFromUrl(): void {
         this.config = WindowUrl.fromUrl(window.location.search);
-        const tileMatrix = this.config.tileMatrix;
         const location = this.getLocationFromHash();
-
-        const loc = this.locationFromLonLat(location);
         const tileGrid = getTileGrid(this.config.tileMatrix.identifier);
-
-        const view = new View({
-            projection: tileMatrix.projection.toEpsgString(),
-            center: loc,
-            zoom: location.zoom,
-            ...tileGrid.getView(),
-        });
-
-        const source = tileGrid.getSource(this.config);
-
-        source.addEventListener('tileloadstart', this.trackTileLoad);
-        source.addEventListener('tileloadend', this.trackTileLoad);
-
-        const layer = new TileLayer({ source });
-
-        this.map = new OlMap({
-            controls: defaultControls({ attributionOptions: { collapsed: false, collapsible: false } }),
-            target: this.el,
-            view,
-            layers: [layer],
-            keyboardEventTarget: document,
+        const style = tileGrid.getStyle(this.config);
+        const coordinate = locationTransform(location, this.config.tileMatrix, MapboxTms);
+
+        this.map = new mapboxgl.Map({
+            container: 'map',
+            style,
+            center: coordinate, // starting position [lon, lat]
+            zoom: location.zoom, // starting zoom
+            attributionControl: false,
         });
 
-        OlAttribution.init(source, this.map, this.config);
-
-        this.map.addEventListener('postrender', this.postRender);
-    }
-
-    tileTrackTimer: unknown | null = null;
-    trackTileLoad = (evt: BaseEvent): boolean => {
-        if (!isTileLoadEvent(evt)) return true;
-        const metricName = 'tile:' + evt.tile.getTileCoord().join('-');
-        if (evt.type === 'tileloadstart') {
-            this.tileTimer.set(metricName, Date.now());
-            return true;
-        }
+        if (tileGrid.tileMatrix.identifier !== GoogleTms.identifier) this.map.setMaxBounds([-180, -85.06, 180, 85.06]);
```

Comment:
can we pull these hard coded values from the GoogleTms.extent or something?

Reply:
Not sure how to make this work now. leave it for next time.